### PR TITLE
Basic pre-playtest prototype

### LINKED ISSUE #71: Refactor FPP networking code

### PR DESCRIPTION
**Pull request by eli-lam**
Changes in this PR:

- Updated all previous network related components to adjust for new main scene and player prefab
- Created new structs for network message passing
- Synchronized animations
- Underground system with respawn (sans rope)

Basically everything that was already implemented for the FPP but now with animations and the underground

Resolves #71

### CODE REVIEW

File: Assets/Prefabs/Tiling/HexTile (Network).prefab [hunk 1]
```diff
@@ -45,9 +45,20 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: af80d41242d914c468ab3d7a97ccbbe0, type: 3}
```
Comment (danielwiktorczyk) [inline Assets/Prefabs/Tiling/HexTile (Network).prefab:45]: Can you change this to `Tile (Network)`? Just for consistency, there is no "HexTile" counterpart

File: Assets/Scripts/Networking/Match/NetworkTile.cs [hunk 2]
```diff
@@ -3,23 +3,16 @@
...
-{
-    public float TimeToRespawn;
-    public float TimeOfBreakingAnimation;
-    public float Progress;
-    public int XIndex;
-    public int ZIndex;
-    public TileState TileState;
-}
-
 public class NetworkTile : MonoBehaviour
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Networking/Match/NetworkTile.cs:6]: This code duplication frightens me. But, if it's the only way to go 😿

**Review by danielwiktorczyk — APPROVED** [review]
Mostly duplicated code, but at least you didn't modify any existing normal code. 

Looks good!

**Review by zhengHugo — APPROVED** [review]
LGTM

File: Assets/Scripts/Networking/Match/NetworkTile.cs (hunk 2 above)
Comment (eli-lam) [reply Assets/Scripts/Networking/Match/NetworkTile.cs:6]: It's mostly so I don't step over you guys's toes and you don't get confused with changes that I'm making

**Pull request merged**